widget/embedded/src/utils/common.ts

```diff
@@ -41,6 +41,22 @@ export function debounce(fn: Function, time: number) {
   }
 }
 
+export function createCustomEvent<T extends Record<string, unknown>>(
+  eventProps?: T
+): CustomEventPayload<T> {
+  let defaultPrevented = false;
+
+  return {
+    preventDefault() {
+      defaultPrevented = true;
+    },
+    get defaultPrevented() {
+      return defaultPrevented;
+    },
+    ...eventProps,
+  } as CustomEventPayload<T>;
```

Comment:
Generally, assertion shouldn't be used and be solved in type itself (except exceptions). The problem is `eventProps` can be undefined. I think that could be solved in a way but if you do the previous change (`emitPreventableEvent`) I requested, I think this will go away from here.  So I just pointed at to make sure we are considering it for future PRs.